receiver/hostmetricsreceiver/internal/scraper/processscraper/process.go

```diff
@@ -0,0 +1,54 @@
+// Copyright 2020, OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package processscraper
+
+import (
+	"github.com/shirou/gopsutil/cpu"
+	"github.com/shirou/gopsutil/process"
+)
+
+type processHandle interface {
+	GetPid() int32
+	Name() (string, error)
+	Exe() (string, error)
+	Username() (string, error)
+	Cmdline() (string, error)
+	Times() (*cpu.TimesStat, error)
+	MemoryInfo() (*process.MemoryInfoStat, error)
+	IOCounters() (*process.IOCountersStat, error)
+}
+
+type gopsProcessHandle struct {
+	*process.Process
+}
+
+func (p *gopsProcessHandle) GetPid() int32 {
+	return p.Pid
+}
+
+var getProcessHandles = getProcessHandlesInternal
```

Comment:
This var is a global state, which makes modifications unsafe e.g. if executing tests in parallel.
I'd recommend making it a field in the `scraper` struct, which is set to `getProcessHandlesInternal` by default, but can be overriden in tests.

Reply:
While I do get your point, I feel like storing this in the scraper struct is a little bit less readable than storing it right above the "real" function here, and I'm not totally sure its worth the benefit of making modifications safe, given that it would only ever be overridden in tests (which don't run in parallel by default anwyay)

But I've made this change anyway 😄